test/y2storage/devicegraph_test.rb

```diff
@@ -197,20 +202,41 @@ def with_sda2_deleted(initial_graph)
         expect(graph.disk_devices).to all(be_a(Y2Storage::Device))
       end
 
-      it "includes all disks and DASDs" do
-        expect(graph.disk_devices.map(&:name)).to contain_exactly(
-          "/dev/dasda", "/dev/dasdb", "/dev/sda", "/dev/sdb", "/dev/sdc",
-          "/dev/sdd", "/dev/sde", "/dev/sdf", "/dev/sdg", "/dev/sdh"
-        )
+      it "includes all disks and DASDs sorted by name" do
+        expect(graph.disk_devices.map(&:name)).to eq [
+          "/dev/dasda", "/dev/dasdb", "/dev/nvme0n1", "/dev/sda", "/dev/sdb",
+          "/dev/sdc", "/dev/sdd", "/dev/sdf", "/dev/sdh", "/dev/sdaa"
+        ]
+      end
+
+      context "even if Disk.all and Dasd.all return unsorted arrays" do
+        before do
+          allow(Y2Storage::Disk).to receive(:all) do |devicegraph|
+            # Let's shuffle things a bit
+            Y2Storage::Partitionable.all(devicegraph).select { |i| i.is?(:disk) }.shuffle
```

Comment:
NP: Maybe it would be better to avoid `#shuffle` usage, because this makes the test runs with "different" input each time.